Modify autoreport to display demands sector-wise

mahendrark opened the pull request:
Fix #535 

The two commands I used to run the simulations and generate the reports:

`mvs_tool -i tests/inputs -ext csv -f -pdf`

`mvs_tool -i tests/benchmark_test_inputs/AFG_grid_heatpump_heat -ext csv -f -pdf`

**Changes proposed in this pull request**:
- Divide the demands section of the autoreport sector-wise

The following steps were realized, as well (if applies):
- [x] Use in-line comments to explain your code
- [x] Write docstrings to your code ([example docstring](https://multi-vector-simulator.readthedocs.io/en/latest/Developing.html#format-of-docstrings))
- [ ] For new functionalities: Explain in readthedocs
- [ ] Write test(s) for your new patch of code (pytests, assertion debug messages)
- [x] Update the CHANGELOG.md
- [x] Apply black (`black . --exclude docs/`)
- [x] Check if benchmark tests pass locally (`EXECUTE_TESTS_ON=master pytest`)


<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl-institut/multi-vector-simulator/blob/dev/CONTRIBUTING.md).*<sub>

mahendrark commented:
@smartie2076, @Bachibouzouk 

I redid everything in a new pull request. This one has the appropriate modifications of different functions and was more intuitive for me than working on the older PR.

Yet, the issue of the heat sector table and plots not being shown in autoreport persists. I tried and tried, but could not find why that issue is occurring. Need your help 

Somehow the function `convert_demand_to_dataframe` defined from line 557 in E1 returns an empty dataframe for heat sector.

mahendrark commented:
I had to do this to keep track of which function is calling which other one xD

![Slide1](https://user-images.githubusercontent.com/13640220/99391430-7b353c00-28da-11eb-8d53-e480156a1ee1.jpg)

smartie2076 commented:
Hi @mahendrark! Will look at this in time. I am however not the person best suited to help you with the timeout problems...

Please make sure that you keep following list updated - it makes it much easier to understand what you did in the PR:

> **Changes proposed in this pull request**:
> - Your_changes

Also, before asking for an official review (when changing from draft to normal PR), you should work through following checkpoints:

> The following steps were realized, as well (if applies):
> - [ ] Use in-line comments to explain your code
> - [ ] Write docstrings to your code ([example docstring](https://multi-vector-simulator.readthedocs.io/en/latest/Developing.html#format-of-docstrings))
> - [ ] For new functionalities: Explain in readthedocs
> - [ ] Write test(s) for your new patch of code (pytests, assertion debug messages)
> - [ ] Update the CHANGELOG.md
> - [ ] Apply black (`black . --exclude docs/`)
> - [ ] Check if benchmark tests pass locally (`EXECUTE_TESTS_ON=master pytest`)
>

smartie2076 commented:
Also, can you please upload a pdf of the current auto-generated report? It is quicker to review this when. I do not have to download it etc

mahendrark commented:
@smartie2076 

I am uploading two simulation reports: Norway harbor (no heat) and the benchmark test report. The benchmark report does not have heat plot and the table is not filled, but it has headers. Also resources are missing (which got printed in the Norway harbor pilot report)

[simulation_report_norway.pdf](https://github.com/rl-institut/multi-vector-simulator/files/5559792/simulation_report_norway.pdf)

[simulation_report.pdf](https://github.com/rl-institut/multi-vector-simulator/files/5559813/simulation_report.pdf)

mahendrark commented:
@smartie2076 

Its ready! Everything works fine now :rocket: :partying_face:

Bachibouzouk commented on src/multi_vector_simulator/E1_process_results.py:
```diff
@@ -582,25 +600,27 @@ def convert_demand_to_dataframe(dict_values):
                 dem: [
                     demands[dem][UNIT],
-                    demands[dem][ENERGY_VECTOR],
+                    demands[dem]["energyVector"],
```
```suggestion
                    demands[dem][ENERGY_VECTOR],
```

Bachibouzouk commented on src/multi_vector_simulator/F1_plotting.py:
```diff
@@ -78,6 +78,15 @@ def convert_plot_data_to_dataframe(plot_data_dict, data_type):
         timeseries for plotting
     """
+
+    # # Modify plot_data_dict to remove non-sector demands
```
@mahendrark  - why do you commit commented code?

Bachibouzouk commented:
@mahendrark - I refactored your 19 come and go commits into 4 conceptual steps (added you as a co-author). To do so I pulled the latest version of `dev` locally, then I move to the PR branch and rebased it on dev `git rebase dev -i` (I dropped the commit @SabineHaas had pushed on `dev` which came from a previous rebase of yours I guess). Then I destroyed the 19 commits with `git reset --soft dev` and I recreated them by adding "Co-authored-by: mahendra <rkmahendranath@gmail.com>" in the long description of commit messages. If you do it, then you stay the author of the refactored commits. Now your PR is way easier to review for someone external.

Bachibouzouk commented:
@mahendrark - please indicate which command line you typed to generate the reports, so that a reviewer can check locally as well. I tested lint and pytest, they work ;)

Bachibouzouk commented on src/multi_vector_simulator/E1_process_results.py:
```diff
@@ -554,24 +554,42 @@ def add_info_flows(settings, dict_asset, flow):
 
-    # Removing all columns that are not actually from demands
+    # Make a dict which is a sub-dict of the JSON results file with only the consumption components of the energy system
+    demands = dict(dict_values[ENERGY_CONSUMPTION])
```
you don't want to use `copy.deepcopy`?

mahendrark replied:
The folks at stackoverflow suggested this on one of the threads, so I used this. Can modify, but it seems to work fine

Bachibouzouk replied:
Stackoverflow is great for patches but can also lead to bugs which are super hard to find in the future. I know deep copy allocates memory for a new copy of the dict, the other method we would need to test it. I changed it (and checked the report was similar)

mahendrark commented:
> 
> 
> @mahendrark - I refactored your 19 come and go commits into 4 conceptual steps (added you as a co-author). To do so I pulled the latest version of `dev` locally, then I move to the PR branch and rebased it on dev `git rebase dev -i` (I dropped the commit @SabineHaas had pushed on `dev` which came from a previous rebase of yours I guess). Then I destroyed the 19 commits with `git reset --soft dev` and I recreated them by adding `Co-authored-by: mahendra <rkmahendranath@gmail.com>` in the long description of commit messages. If you do it, then you stay the author of the refactored commits. Now your PR is way easier to review for someone external.

Thank you! I will use this procedure in the next PRs to condense too many commits into the meaningful few 😎

mahendrark commented:
> 
> 
> @mahendrark - please indicate which command line you typed to generate the reports, so that a reviewer can check locally as well. I tested lint and pytest, they work ;)

The two commands I used to run the simulations and generate the reports:

`mvs_tool -i tests/inputs -ext csv -f -pdf`

`mvs_tool -i tests/benchmark_test_inputs/AFG_grid_heatpump_heat -ext csv -f -pdf`
